### ISSUE TITLE
Does wasmtime support socket?

### ISSUE THREAD

**Issue by ZW007** [bytecodealliance/wasmtime#2732]
I have a socket cpp application, i want to port it to wasm, so i did:
`wasm32-wasi-clang++  server_c1.cpp -o server_c1Wasi.wasm`
but it says 
>server_c1.cpp:28:21: error: use of undeclared identifier 'socket'
    int serv_sock = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);

I wonder if there is any workaround? Thank you!

**Comment by bjorn3** [comment]
No, wasi doesn't have any networking capabilities yet. cc https://github.com/WebAssembly/WASI/issues/370 and https://github.com/WebAssembly/WASI/pull/312
